runtime/parachains/src/paras_inherent.rs

```diff
@@ -115,16 +115,37 @@ impl WeightInfo for TestWeightInfo {
 	}
 }
 
+// Count the ones of the first bitfield in a slice. Careful, since this only counts the ones of
+// the first bitfield, it may be misleading as to how many ones the other bitfields contain. It
+// should only be used in circumstances where an estimate is ok.
+fn bitfields_count_ones(bitfields: &[UncheckedSignedAvailabilityBitfield]) -> usize {
+	if bitfields.len().is_zero() {
+		0
+	} else {
+		bitfields
+			.get(0)
+			.expect("checked for empty bitfield. qed.")
+			.unchecked_payload()
+			.0
+			.count_ones()
+	}
+}
+
 // Note that this does a storage read.
 fn paras_inherent_total_weight<T: Config>(
 	backed_candidates: &[BackedCandidate<<T as frame_system::Config>::Hash>],
 	bitfields: &[UncheckedSignedAvailabilityBitfield],
 	disputes: &[DisputeStatementSet],
-	config: HostConfiguration<T::BlockNumber>,
+	config: &HostConfiguration<T::BlockNumber>,
 ) -> Weight {
-	backed_candidates_weight::<T>(backed_candidates, config)
+	backed_candidates_weight::<T>(backed_candidates)
 		.saturating_add(signed_bitfields_weight::<T>(bitfields.len()))
 		.saturating_add(dispute_statements_weight::<T>(disputes))
+		// in order to get a constant time calculation we get the votes in the first bitfield
+		// and assume the worst case for enacting that many candidates. This may not always be
+		// perfect but should be close enough most of the time for getting an estimate when queuing
+		// extrinsics for block authoring.
+		.saturating_add(enact_candidates_weight::<T>(bitfields_count_ones(bitfields), config))
```

Comment:
Its a bit tricky to get a great estimate upfront because we need to predict which candidates will voted on successfully by looking at the availability bitfields. To be precise, we could iterate through the bitfields, track and the votes and see which will will likely be enacted based on pending available. However that adds a bit of complexity and we already are tight on time from what I understand. My proposal is just to check the votes in the first bitfield and assume the rest are the same. From my understanding this should be ok (assuming the first bitfield is not malicious, which I don't think we can but _hopefully_ that doesn't happen very often). 

Not sure though if I am missing a detail here though

Reply:
We should make sure the bitfields are verified before we do the checking.

This is part of a larger refactor, to integrate weight calculation, caching passed checks, and then doing a random pick all in one `fn`. This is out of scope for this PR.

Reply:
Do you suggest removing this from the total weight calc for now? Or leave as is and create a follow up issue to address the fact that the bitfields are unchecked?

Reply:
(Also in `paras_inherent_total_weight` we need the same data from the unchecked bitfields)